### ISSUE TITLE
All code should have method and class level pydoc strings

### ISSUE THREAD

**Issue by tomoinn** [OpenPiRobotics/CVzero#23]
The cvzero.Tracker class has no docstrings. Any code in this project must be fully documented, so can whoever contributed it please do that?

**Comment by raspibotics** [comment]
Me and vishal have worked on the tracker class. Where do we need to document it?

**Comment by ukBaz** [comment]
Are the following examples helpful:
https://sphinx-rtd-tutorial.readthedocs.io/en/latest/docstrings.html

**Comment by tomoinn** [comment]
Yup, if you do that you'll end up with proper docs for the class being produced when you run 'make html' in the src/docs folder. Sphinx (and RST in general) is picky about indentation and spacing but the doc build will tell you if you're getting stuff wrong!

**Comment by tomoinn** [comment]
You can also look at the code in my libraries, in general everything's heavily documented, i.e. https://github.com/tomoinn/python-pathfinder-tools/blob/master/src/python/mapmaker/__init__.py

**Comment by tomoinn** [comment]
Some useful things you can do - sphinx understands classes and methods and can create links to them, i.e. :class:\`cvzero.Tracker\` will create a link to the generated docs for the tracker class, and :meth:\`cvzero.Tracker.something\` will link directly to the 'something' method in it. If you put a '~' before the name of the thing it'll create a short link (which is handy for me in my libraries as otherwise you get a bunch of links looking like `approxeng.input.foo.blah.Class.method` rather than `Class.method` (in this case I'd use :meth:\`~approxeng.input.foo.blah.Class.method\`)

**Comment by tomoinn** [comment]
Basically you can put anything in the docstring that you could put in a .rst file in the main docs folder. Restructured text isn't the simplest to learn, but it's very powerful (other than tables. Don't try to do tables, they're awful). It's extensible, so you can embed youtube videos (for example) or LaTeX maths, or graphviz format digraphs...
